src/server/src/main/java/io/cassandrareaper/ReaperApplication.java

```diff
@@ -233,11 +234,15 @@ private IStorage initializeStorage(ReaperApplicationConfiguration config, Enviro
       storage = new MemoryStorage();
     } else if ("cassandra".equalsIgnoreCase(config.getStorageType())) {
       storage = new CassandraStorage(config, environment);
-    } else if ("database".equalsIgnoreCase(config.getStorageType())) {
+    } else if ("postgres".equalsIgnoreCase(config.getStorageType()) || "h2".equalsIgnoreCase(config.getStorageType())) {
       // create DBI instance
-      DBI jdbi;
       final DBIFactory factory = new DBIFactory();
-      jdbi = factory.build(environment, config.getDataSourceFactory(), "postgresql");
+
+      DBI jdbi = factory.build(
+              environment,
+              "postgres".equalsIgnoreCase(config.getStorageType())
+                      ? config.getPostgresDataSourceFactory() : config.getH2DataSourceFactory(),
+              "postgresql");
```

Comment:
Can the `DBI` and the `DataSourceFactory` variables be provided by IoC? (instead of inline conditional logic)

Reply:
Good point. Fixed.